prow/gerrit/client/client.go

```diff
@@ -346,6 +347,34 @@ func (c *Client) QueryChangesForInstance(instance string, lastState LastSyncStat
 	return h.queryAllChanges(lastStateForInstance, rateLimit)
 }
 
+// QueryChangesForProject queries change for a project.
+//
+// Important: this method does not update LastSyncState as it is per instance
+// based. It doesn't make sense to update the state as this method has no idea
+// whether all other projects have been queries or not yet. So caller of this
+// method is responsible for making sure that LastSyncState is up-to-date.
+func (c *Client) QueryChangesForProject(instance, project string, lastUpdate time.Time, rateLimit int, additionalFilters []string) ([]ChangeInfo, error) {
+	log := logrus.WithContext(context.Background()).WithField("instance", instance)
+
+	c.lock.RLock()
+	defer c.lock.RUnlock()
+	h, ok := c.handlers[instance]
```

Comment:
the handlers for each instance are updated in place, so imo the update should be avoided while being used to avoid unexpected racing issue. Performance-wise we should be fine since parallel querying is not blocked by read lock